pkg/network/setup/podnic.go

```diff
@@ -91,8 +91,10 @@ func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, iface *v1.In
 func (l *podNIC) PlugPhase2(domain *api.Domain) error {
 	precond.MustNotBeNil(domain)
 
-	if err := l.domainGenerator.Generate(); err != nil {
-		log.Log.Reason(err).Critical("failed to create libvirt configuration")
+	if l.domainGenerator != nil {
```

Comment:
If it is not possible to get to this issue using a normal flow, wouldn't it be safer to return an error? I think that the caller should know that it called `PlugPhase2` on an uninitialized `podNIC`.

Reply:
yes. good idea.